doculab/docs/api-duplicate-prevention.textile

```diff
@@ -0,0 +1,59 @@
+When making API requests, it is possible to receive an error even though your request actually completed successfully.
+
+For example, if you receive a @502 Bad Gateway@ or @504 Gateway Timeout@ error, you can't be sure whether your request was received by Chargify, or not.
+
+If you re-try the request after receiving one of these errors, you _might_ end up with a duplicate transaction.
+
+h2. Uniqueness Token
+
+In order to prevent these duplicates, Chargify allows you to supply a @uniqueness_token@ parameter in any POST or PUT request.
+
+The value you supply for the @uniqueness_token@ parameter is up to you.  We recommend using a long, random value, like a UUID.
+
+If a subsequent request with the same @uniqueness_token@ is received within 60 minutes, it will be rejected with a @409 Conflict@ response code and a duplicate error message.
+
+h2. Example
+
+For example, suppose you are making an adjustment on a subscription.  Using curl, you send the following POST request, including a @uniqueness_token@.
+
+<pre><code>
+curl --verbose -u $CHARGIFY_API_KEY:x -H Accept:application/json -H Content-Type:application/json -X POST \
+-d @adjustment.json https://$CHARGIFY_SUBDOMAIN.chargify.com/subscriptions/$SUBSCRIPTION_ID/adjustments.json
+
+adjustment.json:
+    {"adjustment":
+        {
+          "amount": "-12.43",
+          "memo": "Credit for outage on 1/31"
+        },
+        "uniqueness_token": "2731FB23-98AD-4489-BAF6-7D5CE916F766"
+    }
+</code></pre>
+
+After you send your request, there is some problem, and you receive a @502 Bad Gateway@ error instead of the @201 Created@ response you were hoping for.
```

Comment:
Again, let's remove the reference to a 502 here and just indicate "No response is received or your network request times out without a valid response"